src/Web.cpp

```diff
@@ -925,39 +925,56 @@ void explorerHandleListRequest(AsyncWebServerRequest *request) {
 		return;
 	}
 
-	File file = root.openNextFile();
+	#if defined(HAS_FILEEXPLORER_SPEEDUP) || (ESP_ARDUINO_VERSION_MAJOR == 2 && ESP_ARDUINO_VERSION_MINOR == 0 && ESP_ARDUINO_VERSION_PATCH >= 8)
+		bool isDir = false;
+		String MyfileName = root.getNextFileName(&isDir);
+		while (MyfileName != "") {
+			// ignore hidden folders, e.g. MacOS spotlight files
+			if (!startsWith( MyfileName.c_str() , (char *)"/.")) {
+				JsonObject entry = obj.createNestedObject();
+				convertAsciiToUtf8(MyfileName.c_str(), filePath);
+				std::string path = filePath;
+				std::string fileName = path.substr(path.find_last_of("/") + 1);
+				entry["name"] = fileName;
+				entry["dir"].set(isDir);
+			}
+			MyfileName = root.getNextFileName(&isDir);
+		}
+    #else
+		File file = root.openNextFile();
 
-	while (file) {
-		// ignore hidden folders, e.g. MacOS spotlight files
-		#if ESP_ARDUINO_VERSION_MAJOR >= 2
-		if (!startsWith( file.path() , (char *)"/.")) {
-		#else
-		if (!startsWith( file.name() , (char *)"/.")) {
-		#endif
-			JsonObject entry = obj.createNestedObject();
+		while (file) {
+			// ignore hidden folders, e.g. MacOS spotlight files
 			#if ESP_ARDUINO_VERSION_MAJOR >= 2
-				convertAsciiToUtf8(file.path(), filePath);
+				if (!startsWith( file.path() , (char *)"/.")) {
```

Comment:
This block could be made more readable by using a const variable (instead of fetching the file path twice) and then you can use the same code as in the loop in Line 934:

```cpp
			// ignore hidden folders, e.g. MacOS spotlight files
			const String MyfileName;
			#if ESP_ARDUINO_VERSION_MAJOR >= 2
				MyfileName = file.path();
			#else
				MyfileName = file.name();
			#endif
			if(!startsWith(MyfileName, "/:")) {
				JsonObject entry = obj.createNestedObject();
				convertAsciiToUtf8(MyfileName.c_str(), filePath);
				// and so on
			}
```